packages/hardhat-kadena/README.md

```diff
@@ -405,6 +405,186 @@ npx hardhat print-config
+      },
+    },
+  },
+};
+```
+
+You can use several "in-process" chainweb config for testing your smart contract behavior against different chain numbers or graphs or any other configuration
+
+```typescript
+module.exports = {
+  solidity: '0.8.28',
+  chainweb: {
+    // use 2 chains for regular development so tests run fast
+    hardhat: {
+      chains: 2,
+    },
+    // use 5 chains with a custom graph that is close to testnet
+    semiTestnet: {
+      chains: 20,
+    },
+    // use 3 chains with a custom graph
+    nonStandardGraph: {
+      chains: 3,
+      graph: {
+        0: [1],
+        1: [2],
+        2: [0],
+      },
+    },
+  },
+};
+```
+
+#### Research Graph behaviors
+
+Graph has direct impact on the security and performance of the chainweb is also might change the SPV proof creation and verification time, so you can use this option for more advanced scenarios.
+
+```typescript
+module.exports = {
+  chainweb: {
+    // this config uses 4 in-process chains.
+    custom_graph: {
+      chains: 4,
+      // using a custom graph to see the network behavior
+      graph: {
+        0: [1],
+        1: [0, 2, 3],
+        2: [0],
+        3: [1, 2, 0],
+      },
+    },
+  },
+};
+```
+
+### **How to use advance hardhat configs**
+
+All in-process networks inherit the default hardhat network configuration so if for example you want to set `allowUnlimitedContractSize` you just need to add it to the config
+
+```typescript
+module.exports = {
+  solidity: '0.8.28',
+  networks: {
+    // then both hardhat and semiTestnet chainweb uses this config since both are internal
+    hardhat: {
+      allowUnlimitedContractSize: true,
+      fork: {
+        url:''
+      }
+    }
+  }
+  chainweb: {
+    hardhat: {
+      chains: 2,
+    },
+    semiTestnet: {
+      chains: 20,
+    },
+  },
+};
+```
+
+### **What is the difference between `in-process` and `external` Chainweb types?**
+
+- **`in-process`**: Runs Chainweb as part of Hardhat's local network, ideal for development and testing.
+- **`external`**: Connects to an existing Chainweb network (e.g., testnet or mainnet) using an external RPC.
+
+### **How do I switch between different Chainwebs?**
+
+Use the `--chainweb` flag in your Hardhat commands:
+
+```bash
+npx hardhat test --chainweb devnet
+```
+
+Or manually set the `defaultChainweb` in your config.
+
+### **How do I deploy a contract to all chains in my Chainweb setup?**
+
+Use the `deployContractOnChains` function:
+
+```ts
+await chainweb.deployContractOnChains('SimpleToken');
+```
+
+This will deploy the contract to all available chains.
+
+### **How can I request an SPV proof for a cross-chain transaction?**
+
+Use the `requestSpvProof` function:
+
+```ts
+const proof = await chainweb.requestSpvProof(targetChain, origin);
+```
+
+### **Can I customize the Chainweb chain connection graph?**
+
+Yes, you can define a custom graph in your configuration:
+
+```ts
+chainweb: {
+  customGraph: {
```

Comment:
Same as above, it might be handy to all this `customGraphChainweb` to make it clear it is a whole chainweb config is and not just a graph config.